buildSrc/src/main/java/org/elasticsearch/gradle/test/DistroTestPlugin.java

```diff
@@ -100,58 +88,101 @@ public void apply(Project project) {
 
         // TODO: it would be useful to also have the SYSTEM_JAVA_HOME setup in the root project, so that running from GCP only needs
         // a java for gradle to run, and the tests are self sufficient and consistent with the java they use
+        NamedDomainObjectContainer<ElasticsearchDistribution> allDistributions = DistributionDownloadPlugin.getContainer(project);
+        List<ElasticsearchDistribution> testDistributions = configureDistributions(project);
 
-        Version upgradeVersion = getUpgradeVersion(project);
-        Provider<Directory> distributionsDir = project.getLayout().getBuildDirectory().dir("packaging/distributions");
-        Provider<Directory> upgradeDir = project.getLayout().getBuildDirectory().dir("packaging/upgrade");
-
-        List<ElasticsearchDistribution> distributions = configureDistributions(project, upgradeVersion);
-        TaskProvider<Copy> copyDistributionsTask = configureCopyDistributionsTask(project, distributionsDir);
-        TaskProvider<Copy> copyUpgradeTask = configureCopyUpgradeTask(project, upgradeVersion, upgradeDir);
-
-        Map<ElasticsearchDistribution.Type, TaskProvider<?>> lifecyleTasks = lifecyleTasks(project, "destructiveDistroTest");
+        Map<ElasticsearchDistribution.Type, TaskProvider<?>> lifecycleTasks = lifecycleTasks(project, "destructiveDistroTest");
+        Map<String, TaskProvider<?>> versionTasks = versionTasks(project, "destructiveDistroUpgradeTest");
         TaskProvider<Task> destructiveDistroTest = project.getTasks().register("destructiveDistroTest");
 
         Configuration examplePlugin = configureExamplePlugin(project);
 
-        for (ElasticsearchDistribution distribution : distributions) {
-            TaskProvider<?> destructiveTask = configureDistroTest(project, distribution, dockerSupport, examplePlugin);
+        List<TaskProvider<Test>> windowsTestTasks = new ArrayList<>();
+        Map<Type, List<TaskProvider<Test>>> linuxTestTasks = new HashMap<>();
+        Map<String, List<TaskProvider<Test>>> upgradeTestTasks = new HashMap<>();
+        Map<String, TaskProvider<?>> depsTasks = new HashMap<>();
+        for (ElasticsearchDistribution distribution : testDistributions) {
+            String taskname = destructiveDistroTestTaskName(distribution);
+            TaskProvider<?> depsTask = project.getTasks().register(taskname + "#deps");
```

Comment:
I don't think we want the VM task depending on all distributions. That would mean a repro line would rebuild all distros, instead of just the one you are trying to run.

Reply:
Ok, I got confused by the fact we are passing all the deps tasks to the wrapper task creation. Couldn't we do the same thing though, and directly bucket the distributions, instead of the deps tasks? Just seems another layer we don't really need.

Reply:
We would need to apply the distro download plugin inside the subprojects (ie the ones with the wrapper tasks), as well as produce the test jar as an artifact. However, that is a lot of extra work that is not useful since we don't actually need those in the subproject to run, we only want to ensure the tasks in `:qa:os` ran before launching the VM task.